pkg/operator/controller/nodeobservability/daemonset.go

```diff
@@ -31,59 +33,101 @@ const (
 // ensureDaemonSet ensures that the daemonset exists
 // Returns a Boolean value indicating whether it exists, a pointer to the
 // daemonset and an error when relevant
-func (r *NodeObservabilityReconciler) ensureDaemonSet(ctx context.Context, nodeObs *v1alpha1.NodeObservability, sa *corev1.ServiceAccount, ns string) (bool, *appsv1.DaemonSet, error) {
+func (r *NodeObservabilityReconciler) ensureDaemonSet(ctx context.Context, nodeObs *v1alpha1.NodeObservability, sa *corev1.ServiceAccount, ns string) (*appsv1.DaemonSet, error) {
 	nameSpace := types.NamespacedName{Namespace: ns, Name: daemonSetName}
 	desired := r.desiredDaemonSet(nodeObs, sa, ns)
 	if err := controllerutil.SetControllerReference(nodeObs, desired, r.Scheme); err != nil {
-		return false, nil, fmt.Errorf("failed to set the controller reference for daemonset: %w", err)
-	}
-	exist, current, err := r.currentDaemonSet(ctx, nameSpace)
-	if err != nil {
-		return false, nil, fmt.Errorf("failed to get DaemonSet: %w", err)
+		return nil, fmt.Errorf("failed to set the controller reference for daemonset: %w", err)
 	}
-	if !exist {
-		cmExists, err := r.createConfigMap(ctx, nodeObs, ns)
+
+	current, err := r.currentDaemonSet(ctx, nameSpace)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, fmt.Errorf("failed to get daemonset %q due to: %w", nameSpace, err)
+	} else if err != nil && errors.IsNotFound(err) {
+
+		// create daemon since it doesn't exist
+		err := r.createConfigMap(ctx, nodeObs, ns)
 		if err != nil {
-			return false, nil, fmt.Errorf("failed to create the configMap for kubelet-serving-ca: %w", err)
-		}
-		if !cmExists {
-			return false, nil, fmt.Errorf("failed to get the configMap for kubelet-serving-ca: %w", err)
+			return nil, fmt.Errorf("failed to create the configmap for kubelet-serving-ca: %w", err)
 		}
+
 		if err := r.createDaemonSet(ctx, desired); err != nil {
-			return false, nil, err
+			return nil, fmt.Errorf("failed to create daemonset %q: %w", nameSpace, err)
 		}
+		r.Log.V(1).Info("created daemonset", "ds.namespace", nameSpace.Namespace, "ds.name", nameSpace.Name)
+
 		return r.currentDaemonSet(ctx, nameSpace)
 	}
-	return true, current, err
+
+	updated, err := r.updateDaemonset(ctx, current, desired)
+	if err != nil {
+		return nil, fmt.Errorf("failed to update the daemonset %q: %w", nameSpace, err)
+	}
+
```

Comment:
Can you please add a log for the updated daemon set, similar to what you did after the creation?